### PR TITLE
chore: terminology

### PR DESCRIPTION
**Pull request by MatejVukosav**
- Add common terms. 

If there is any term that might be good to explain, feel free to suggest with example.

### CODE REVIEW

File: docs/docs/02-learn/02-terminology.mdx [hunk 1]
```diff
@@ -3,10 +3,12 @@ id: terminology
...
-- Node
-- Peer
-- Application
-- Context
-- Identity
-- Root key
-- Client key
+As projects grow, it's important to have a shared vocabulary to help communicate effectively. This page provides a list of terms used in the documentation and throughout the codebase.
+
+- **Node** is any individual device or computer that participates in the network. To avoid confusion with network layer used in the protocol, instead of network we are using term **Context**.
```
Comment (miraclx) [inline docs/docs/02-learn/02-terminology.mdx:8]: Can you rephrase this line?

It's just a gateway to the Calimero network

File: docs/docs/02-learn/02-terminology.mdx [hunk 2]
```diff
@@ -3,10 +3,12 @@ id: terminology
...
+- **Peer** is a specific instance of a node within a P2P network that interacts with other peers. Peer represents user.
```
Comment (miraclx) [inline docs/docs/02-learn/02-terminology.mdx:9]: Network actor, the identity with which all activity over the network through your node is performed

File: docs/docs/02-learn/02-terminology.mdx [hunk 3]
```diff
@@ -3,10 +3,12 @@ id: terminology
...
+- **Context** is the core of the Calimero ecosystem. It is an application specific network designed to enable direct communication between users, eliminating the need for intermediaries
+- **Application** is a software program designed to perform specific tasks or solve particular problems. To ensure compatibility and functionality, it should be developed according to the protocol SDK instructions provided. Once developed, the application should be published in a format that others can use during runtime, specifically in WebAssembly (WASM) format. An application can also include a frontend component, allowing users to interact with it directly. This user interface facilitates interaction with the underlying software, making the application accessible and user-friendly.
```
Comment (miraclx) [inline docs/docs/02-learn/02-terminology.mdx]: The application "includes" the frontend component?

File: docs/docs/02-learn/02-terminology.mdx (hunk 1 above)
Comment (MatejVukosav) [reply docs/docs/02-learn/02-terminology.mdx:8]: Which one?
Gateway is too vague and not accurate

File: docs/docs/02-learn/02-terminology.mdx (hunk 2 above)
Comment (MatejVukosav) [reply docs/docs/02-learn/02-terminology.mdx:9]: Added peer id

File: docs/docs/02-learn/02-terminology.mdx (hunk 3 above)
Comment (MatejVukosav) [reply docs/docs/02-learn/02-terminology.mdx]: Bad phrasing. Changed
